File: pset6/caesar/caesar.py
from cs50 import get_string
import sys

if len(sys.argv) != 2:
    print("Usage: ./caesar k")
    exit(1)

# get the key
key = int(sys.argv[1])

# get the plain_text
plain_str = get_string("plain text: ")
cipher_str = ""

# encipher the text
for j in range(len(plain_str)):
    if plain_str[j].isalpha():
        if plain_str[j].isupper():
            aphi = int(ord(plain_str[j])) - 65
            aphi = (aphi + key) % 26 + 65
        if plain_str[j].islower():
            aphi = int(ord(plain_str[j])) - 97
            aphi = (aphi + key) % 26 + 97
        cipher_str = cipher_str + chr(aphi)
    else:
        cipher_str = cipher_str + plain_str[j]

# print the ciphertext
print(f"ciphertext: {cipher_str}")

File: pset2/caesar/caesar.c
#include <stdio.h>
#include <ctype.h>
#include <cs50.h>
#include <string.h>

int main(int argc, string argv[])
{
    if (argc == 2)
    {
        string num = argv[1];
        int i = atoi(num);
        printf("plaintext: ");
        string s = get_string();
        printf("ciphertext: ");
        for (int j = 0, n = strlen(s); j < n; j++)
        {
            if (isalpha(s[j]))
            {
                if (isupper(s[j]))
                {
                    int aphi = s[j] - 65;
                    printf("%c", ((aphi + i) % 26 + 65));
                }
                if (islower(s[j]))
                {
                    int aphi = s[j] - 97;
                    printf("%c", (aphi + i) % 26 + 97);
                }
            }
            else
            {
                printf("%c", s[j]);
            }
        }
        printf("\n");
    }
    else
    {
        printf("Usage: ./caesar k\n");
        return 1;
    }
}